server/s3_action_base.cc

```diff
@@ -57,9 +56,16 @@ void S3Action::set_authorization_meta() { next(); }
 
 void S3Action::check_authorization() {
   s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
-  auth_client->check_authorization(
-      std::bind(&S3Action::check_authorization_successful, this),
-      std::bind(&S3Action::check_authorization_failed, this));
+
+  if (is_authorizationheader_present) {
+    auth_client->check_combo_auth(
```

Comment:
You can put a comment here ie if authorization header present then authentication...authorization....being done....

Reply:
The absence of "Authorization: ..." header means anonymous access.
ACL or policy of an entity can allow anonymous access.
Authorization only is needed in cases of anonymous access